journal/src/test/java/io/camunda/zeebe/journal/file/SegmentsManagerTest.java

```diff
@@ -48,16 +48,16 @@ void shouldDeleteFilesMarkedForDeletionsOnLoad() {
     // when
     // if we close the current journal, it will delete the files on closing. So we cannot test this
     // scenario.
-    final var newSegments = createSegmentsManager(0);
-    newSegments.open();
+    try (var newSegments = createSegmentsManager(0)) {
+      newSegments.open();
```

Comment:
:x: From the comment, I expected the assertions need to be inside this right?

Reply:
Oops, yeah you are right :sweat_smile:

Reply:
But actually the test has been broken for longer than that I think. It wants to test that files marked for deletion are deleted **when opening**. So the initial segments manager must not be closed, otherwise the files are deleted **when closing**. I think this

Reply:
I think the test was actually broken before. I've adjusted and updated the comments to clarify what's being tested here.